src/app/src/main/java/com/example/bookmark/ManageRequestsActivity.java

```diff
@@ -9,9 +25,50 @@
  */
 public class ManageRequestsActivity extends BackButtonActivity {
 
+    ListView requestList;
+    ArrayAdapter<Request> requestAdapter;
+    ArrayList<Request> requestDataList;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_manage_requests);
+
+        // TODO: make sure this matches with what Konrad is doing
+        SharedPreferences mPrefs = getSharedPreferences("Key", MODE_PRIVATE);
+        String username = mPrefs.getString("username", null);
+
+        Intent intent = getIntent();
+        Bundle bundle = intent.getExtras();
+        Book book = (Book) bundle.getSerializable("Book");
+
+        TextView bookName = findViewById(R.id.book_name);
+        bookName.setText(book.getTitle());
+
+        requestList = findViewById(R.id.request_list);
+        requestDataList = new ArrayList<>();
+        requestAdapter = new RequestList(this, requestDataList);
+        requestList.setAdapter(requestAdapter);
+
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        final CollectionReference collectionReference = db.collection("Requests");
+
+        collectionReference.addSnapshotListener((queryDocumentSnapshots, e) -> {
+
+            requestDataList.clear();
+            for(QueryDocumentSnapshot doc: queryDocumentSnapshots) {
+                // TODO: clarify once Firebase is working as expected
+                Owner requestOwner = (Owner) doc.getData().get("Owner");
+                Book requestBook = (Book) doc.getData().get("Book");
+                if (requestOwner.getUsername().equals(username) && requestBook.equals(book)) {
+                    Borrower borrower = (Borrower) doc.getData().get("Requester");
+                    String requestDate = doc.getData().get("Request Date").toString();
+                    requestDataList.add(new Request(requestBook, borrower, null));
+                }
+            }
+            requestAdapter.notifyDataSetChanged();
+        });
```

Comment:
Want me to add a method to FirebaseProvider that allows you to subscribe to requests? Or would a method that just gets requests from the database be sufficient?

Reply:
I think with the addSnapshotListener within this file, a method that just gets requests is sufficient but if this is being needed by other files then it could be useful to implement here

Reply:
The problem with this, at least as currently implemented, is that we technically have infinite nesting/recursion. Specifically, these two lines are problematic:
```
                Owner requestOwner = (Owner) doc.getData().get("Owner");
                Book requestBook = (Book) doc.getData().get("Book");
```
An owner contains Book objects. A book contains its Owner. Our database can't possibly represent this without decoupling the two objects. I'm changing Book to only store the username of its owner to try and break this infinite nesting/recursion. Other entities also have this problem and I'm working to address it.